src/journal/journald-kmsg.c

```diff
@@ -261,15 +264,19 @@ void dev_kmsg_record(Server *s, char *p, size_t l) {
         if (asprintf(&syslog_facility, "SYSLOG_FACILITY=%i", LOG_FAC(priority)) >= 0)
                 iovec[n++] = IOVEC_MAKE_STRING(syslog_facility);
 
+        log_max_level = log_get_max_level();
```

Comment:
please move this where you actually override the log level, i.e. right before the log_set_max_level() in the if condition below.